internal/providers/qemu/qemu_fwcfg.go

```diff
@@ -34,22 +38,74 @@ import (
 )
 
 const (
-	firmwareConfigPath = "/sys/firmware/qemu_fw_cfg/by_name/opt/com.coreos/config/raw"
+	firmwareConfigPath     = "/sys/firmware/qemu_fw_cfg/by_name/opt/com.coreos/config/raw"
+	firmwareConfigSizePath = "/sys/firmware/qemu_fw_cfg/by_name/opt/com.coreos/config/size"
 )
 
 func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
+	// load qemu_fw_cfg module
 	_, err := f.Logger.LogCmd(exec.Command("modprobe", "qemu_fw_cfg"), "loading QEMU firmware config module")
 	if err != nil {
 		return types.Config{}, report.Report{}, err
 	}
 
-	data, err := ioutil.ReadFile(firmwareConfigPath)
+	// get size of firmware blob, if it exists
+	sizeBytes, err := ioutil.ReadFile(firmwareConfigSizePath)
 	if os.IsNotExist(err) {
 		f.Logger.Info("QEMU firmware config was not found. Ignoring...")
+		return util.ParseConfig(f.Logger, []byte{})
 	} else if err != nil {
-		f.Logger.Err("couldn't read QEMU firmware config: %v", err)
+		f.Logger.Err("couldn't read QEMU firmware config size: %v", err)
+		return types.Config{}, report.Report{}, err
+	}
+	size, err := strconv.Atoi(strings.TrimSpace(string(sizeBytes)))
```

Comment:
Is there a reasonable size cap limit we should check against here so we're not allocating arbitrarily large buffers?

Reply:
Anything over a couple hundred MB won't complete in a reasonable time, with the current kernel code.  But is it worth imposing a cap here?  The worst that will happen is that the system will OOM and provisioning will fail.

Reply:
Just as a general rule whenever allocating dynamically-sized buffers, I like to have a sane max cap to check against. You're right in the Ignition case, failure is a more reasonable option.